packages/interfaces/src/receiver.ts

```diff
@@ -0,0 +1,17 @@
+import type { IDecodedMessage, IDecoder } from "./message.js";
+import type { Callback, ProtocolOptions } from "./protocols.js";
+
+type Unsubscribe<T> = () => T;
```

Comment:
```suggestion
type Unsubscribe = () => void | Promise<void>;
```

Looks like the template type is always `void`, hence not needed....?

Reply:
Seems right